src/main/java/org/elasticsearch/action/support/nodes/NodesOperationRequest.java

```diff
@@ -43,6 +43,11 @@ protected NodesOperationRequest() {
 
     }
 
+    protected NodesOperationRequest(ActionRequest request, String... nodesIds) {
+        super(request);
+        this.nodesIds = nodesIds;
+    }
+
     protected NodesOperationRequest(String... nodesIds) {
```

Comment:
I wish we could remove it, but it's used by other requests that are created directly through java api, for instance nodes hot threads, nodes info etc. those don't need headers in their constructor, as the client automatically copies over the headers in the Rest action.